core/service/pool.go

```diff
@@ -175,3 +176,22 @@ func (i *Instance) Proposal() market.ServiceProposal {
 func (i *Instance) State() State {
 	return i.state
 }
+
+type instanceSocket struct {
+	Id         string `json:"id"`
+	Status     string `json:"status"`
+	ProviderID string `json:"providerId"`
+	Type       string `json:"type"`
+}
+
+// SetState - Set instance state and send new state to websocket
+func (i *Instance) SetState(state State, socket websocket.WebSocket) {
```

Comment:
What is the purpose of this function?

Reply:
Set instance state and send new state to websocket